tests/post/steps/test_volume.py

```diff
@@ -134,4 +169,38 @@ def _create_volume(host, name, k8s_custom_client):
         if isinstance(exc, ApiException) and exc.status == 404:
             assert False, 'Exception while creating volume {}'.format(name)
         raise
+
+
+def _volume_exist(host, name, k8s_custom_client):
```

Comment:
This return an object, not a Boolean: call it `_get_volume`.

`host` is unused: can be removed.
Here we're not using the fixture (it's not a `then`/`when`/`given`, so fixture are not magically applied): we can name the parameter `k8s_client`).


Also we know that we're fetching a Volume here, so the variable can be named `volume` instead of `custom_object`

```python
 def _get_volume(k8s_client, name):
    try:
        return k8s_custom_client.get_cluster_custom_object(
            group="storage.metalk8s.scality.com",
            version="v1alpha1",
            plural="volumes",
            name=name
        )
    except (ApiException, HTTPError) as exc:
        if isinstance(exc, ApiException) and exc.status == 404:
            return None
        raise
```

Reply:
> This return an object, not a Boolean: call it `_get_volume`.
> 
> `host` is unused: can be removed.
> Here we're not using the fixture (it's not a `then`/`when`/`given`, so fixture are not magically applied): we can name the parameter `k8s_client`).

Not sure this works as stated, tested and it only works with the right fixtures (`k8s_custom_client`). Safe to say that for some reason, fixtures are magically applied here.

> 
> Also we know that we're fetching a Volume here, so the variable can be named `volume` instead of `custom_object`
> 
> ```python
>  def _get_volume(k8s_client, name):
>     try:
>         return k8s_custom_client.get_cluster_custom_object(
>             group="storage.metalk8s.scality.com",
>             version="v1alpha1",
>             plural="volumes",
>             name=name
>         )
>     except (ApiException, HTTPError) as exc:
>         if isinstance(exc, ApiException) and exc.status == 404:
>             return None
>         raise
> ```

Reply:
I applied this diff (which is what I was recommending) on your current PR:

```diff
--- a/tests/post/steps/test_volume.py
+++ b/tests/post/steps/test_volume.py
@@ -160,9 +160,9 @@ def check_pv_absent(name, k8s_client):
 # Helpers {{{
 
 
-def _create_volume(body, k8s_custom_client):
+def _create_volume(body, k8s_client):
     try:
-        k8s_custom_client.create_cluster_custom_object(
+        k8s_client.create_cluster_custom_object(
             group="storage.metalk8s.scality.com",
             version="v1alpha1",
             plural="volumes",
@@ -174,9 +174,9 @@ def _create_volume(body, k8s_custom_client):
         raise
 
 
-def _get_volume(name, k8s_custom_client):
+def _get_volume(name, k8s_client):
     try:
-        return k8s_custom_client.get_cluster_custom_object(
+        return k8s_client.get_cluster_custom_object(
             group="storage.metalk8s.scality.com",
             version="v1alpha1",
             plural="volumes",
@@ -187,9 +187,8 @@ def _get_volume(name, k8s_custom_client):
             return None
 
 
-def _delete_volume(name, k8s_custom_client):
-
-    k8s_custom_client.delete_cluster_custom_object(
+def _delete_volume(name, k8s_client):
+    k8s_client.delete_cluster_custom_object(
         group="storage.metalk8s.scality.com",
         version="v1alpha1",
         plural="volumes",
```

And it works…

So no, fixtures are **not** magically applied :slightly_smiling_face: